operator/v1/types_console.go

```diff
@@ -143,12 +143,32 @@ type ProjectAccess struct {
 	AvailableClusterRoles []string `json:"availableClusterRoles,omitempty"`
 }
 
+// A sub-catalog type cannot be in both `enabled` and `disabled` list.
+type DeveloperConsoleCatalogTypes struct {
```

Comment:
We either need a webhook or CEL (https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules) for validating that an item is in either enabled or disabled but not both

The CEL validation would be for checking two listSets are disjoint
```
self.set1.all(e, !(e in self.set2))
```
I'll work out how this is applied as this is new for K8s 1.25

Reply:
we don't need this if we add state as a unionDiscriminator